tests/vm.py

```diff
@@ -0,0 +1,287 @@
+import string
+import random
+import time
+import os
+import signal
+import subprocess
+from os import path
+from utils import (
+    is_ipv6_configured,
+)
+
+profile = os.environ.get("LXC_PROFILE", "lxc/microk8s.profile")
+
+
+class VM:
+    """
+    This class abstracts the backend we are using. It could be either multipass or lxc.
+    """
+
+    launch_config = """---
+version: 0.1.0
+extraCNIEnv:
+  IPv4_SUPPORT: true
+  IPv4_CLUSTER_CIDR: 10.3.0.0/16
+  IPv4_SERVICE_CIDR: 10.153.183.0/24
+  IPv6_SUPPORT: true
+  IPv6_CLUSTER_CIDR: fd02::/64
+  IPv6_SERVICE_CIDR: fd99::/108
+extraSANs:
+  - 10.153.183.1"""
+
+    def __init__(self, backend=None, attach_vm=None, enable_ipv6=True):
+        """Detect the available backends and instantiate a VM.
+
+        If `attach_vm` is provided we just make sure the right MicroK8s is deployed.
+        :param backend: either multipass of lxc
+        :param attach_vm: the name of the VM we want to reuse
+        :param enable_ipv6: if set, IPv6 support is enabled in the launch config (enabled by default)
+        """
+        rnd_letters = "".join(random.choice(string.ascii_lowercase) for i in range(6))
+        self.backend = backend
+        self.vm_name = "vm-{}".format(rnd_letters)
+        self.attached = False
+        if attach_vm:
+            self.attached = True
+            self.vm_name = attach_vm
+
+        if not enable_ipv6:
```

Comment:
This VM code is copied from `test-cluster.py`.

The only added the flag `enable_ipv6` and this check in the `__init__` method